clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/vmimage/view/AzureVMImageLookupController.groovy

```diff
@@ -61,15 +62,31 @@ class AzureVMImageLookupController {
   List<AzureNamedImage> getVMImage(@PathVariable String account, @PathVariable String region, @PathVariable String imageId) {
     def result = new ArrayList<AzureNamedImage>()
 
+    /// First search for any matches in the custom image cache
+    result.addAll(
+      getAllMatchingKeyPattern(
+        imageId,
+        Keys.Namespace.AZURE_CUSTOMVMIMAGES.ns,
+        Keys.getCustomVMImageKey(azureCloudProvider,
+          account,
+          region,
+          "*"),
+        true))
+
+    if (!result.isEmpty()) {
+      // found at least one match
+      return result
+    }
+
     // return a list of virtual machine images as read from the config.yml file
-    // TODO: retrieve the list of virtual machine images from the azure respective cache
     accountCredentialsProvider.getAll().each { creds ->
       if(creds instanceof AzureNamedAccountCredentials && creds.accountName == account) {
         creds.vmImages.each { vmImage ->
-          def imageName = vmImage.offer + " " + vmImage.sku + " (${vmImage.publisher})"
+          def imageName = vmImage.offer + "-" + vmImage.sku + "(Recommended)"
```

Comment:
Is this the only place this name generation logic is used? It might be helpful to extract into a utility class.